### PR TITLE
CLI application for submitting bulk request to the bulk API endpoint.

### PR DESCRIPTION
**Pull request by kiivihal**
The goal of this pull request is to have a default method to load serialized bulk-requests from disk and send them to the bulk-API endpoint.

* Config initiation is no longer default for cobra based cli application.
* bulk CLI sub-command has been added
* bulk package is extended with a publisher client that can be used as a library.

### CODE REVIEW

File: ikuzo/ikuzoctl/cmd/bulk.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	chunkSize   int
+)
+
+// bulkCmd represents the bulk command
+var bulkCmd = &cobra.Command{
+	Use:   "bulk",
+	Short: "Load bulk.Request from disk",
+	Long: `Loading bulk.Requests serialized as line delimited json.
+
+	We assume that the structure is 'orgID/datasetID/hubID.jsonl' each bulk.Request
```
Comment (iicing) [inline ikuzo/ikuzoctl/cmd/bulk.go:39]:
```suggestion
	We assume that the structure is 'orgID/datasetID/hubID.jsonl' and that each bulk.Request
```

File: ikuzo/ikuzoctl/cmd/bulk.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	is stored in single file.`,
```
Comment (iicing) [inline ikuzo/ikuzoctl/cmd/bulk.go:40]:
```suggestion
	is stored in a single file.`,
```

File: ikuzo/ikuzoctl/cmd/bulk.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+func init() {
+	rootCmd.AddCommand(bulkCmd)
+
+	bulkCmd.Flags().StringVarP(&requestPath, "dataPath", "p", ".", "Full path to orgIDs for the bulk.Requests.")
+	bulkCmd.Flags().StringVarP(&publishHost, "host", "", "http://localhost:3001", "network host of where target hub3 is running")
+	bulkCmd.Flags().IntVarP(&chunkSize, "chunkSize", "", 500, "size of number of records send per batch")
+}
+
+func publish(ctx context.Context, host, dataPath string) error {
```
Comment (iicing) [inline ikuzo/ikuzoctl/cmd/bulk.go:55]: host and dataPath are never used, instead publishHost and requestPath are used but they are never assigned a value

File: ikuzo/service/x/bulk/publisher.go [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+	}
+
+	return p.AppendBytes(b)
+}
+
+// Do parses the records in the dataPath and submits them in chunks to the Hub3 BulkAPI endpoint
+//
+// Chunks are defined by p.BulkSize.
+//
+// The expect directory structure is:
```
Comment (iicing) [inline ikuzo/service/x/bulk/publisher.go:93]:
```suggestion
// The expected directory structure is:
```

File: ikuzo/service/x/bulk/publisher.go [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+				b, err := ioutil.ReadFile(requestPath)
+				if err != nil {
+					return fmt.Errorf("unable to read jsonl file; %w", err)
+				}
+
+				if err := p.AppendBytes(b); err != nil {
+					return err
+				}
+
+				select {
```
Comment (iicing) [inline ikuzo/service/x/bulk/publisher.go:155]: could currently also be an if statement

File: ikuzo/service/x/bulk/publisher.go [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+	defer p.rw.Unlock()
+
+	body := bytes.Join(p.requests, []byte("\n"))
+
+	resp, err := p.post(body)
+	if err != nil {
+		return fmt.Errorf("unable to post to endpoint: %w", err)
+	}
+
+	if resp.StatusCode < 200 || resp.StatusCode > 300 {
```
Comment (iicing) [inline ikuzo/service/x/bulk/publisher.go:188]:
```suggestion
	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
```

File: ikuzo/ikuzoctl/cmd/bulk.go (hunk 3 above)
Comment (kiivihal) [reply ikuzo/ikuzoctl/cmd/bulk.go:54]: I updated the code to not use the global var from the flag but use the variables that are passed in.

**Pull request merged**